tests/ast_pattern_matching.nim

```diff
@@ -133,8 +133,8 @@ proc matchLengthKind*(arg: NimNode; kind: NimNodeKind; length: int): MatchingErr
   matchLengthKind(arg, {kind}, length)
 
 proc matchValue(arg: NimNode; kind: set[NimNodeKind]; value: SomeInteger): MatchingError {.compileTime.} =
-  let kindFail   = not(kind.card == 0 or arg.kind in kind)
-  let valueFail  = arg.intVal != int(value)
+  template kindFail: bool  = not(kind.card == 0 or arg.kind in kind)
+  template valueFail: bool = arg.intVal != int(value)
```

Comment:
Why this change?

Reply:
I introduced these local variables for better readability in my library ast-pattern-matching. There was a problem though with short circuit behavior in the test I introduced. That is a bug in ast-pattern-matching that I wasn't aware of. I also fixed it in that library. It is already pushed there.

Reply:
looks like we can revert these 2 lines then?

Reply:
I don't think that's what krux meant.